build/action.yaml

```diff
@@ -35,18 +40,49 @@ runs:
       uses: andstor/file-existence-action@v3
       with:
         files: yarn.lock
+    - name: Compute lockfile hash
+      if: inputs.s3-bucket-name != ''
+      id: lockfile_hash
+      run: echo "hash=${{ steps.check_yarn_lock.outputs.files_exists == 'true' && hashFiles('**/yarn.lock') || hashFiles('**/package-lock.json') }}" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Load cached node_modules if available
+      if: inputs.s3-bucket-name != ''
+      uses: everpcpc/actions-cache@v1
+      id: read_cache
+      with:
+        bucket: ${{ inputs.s3-bucket-name }}
+        use-fallback: false
+        path: node_modules
+        key: ${{ env.cache-name }}-${{ steps.lockfile_hash.outputs.hash }}
+        restore-keys: ${{ env.cache-name }}-
+      env:
+        AWS_REGION: ${{ inputs.s3-bucket-region }}
+        cache-name: ${{ github.event.repository.name }}-cache-node-modules
     - name: Install dependencies (yarn)
-      if: steps.check_yarn_lock.outputs.files_exists == 'true'
+      if: steps.check_yarn_lock.outputs.files_exists == 'true' && steps.read_cache.outputs.cache-hit != 'true'
       shell: bash
       run: yarn --pure-lockfile
       env:
         NPM_AUTH_TOKEN: ${{ inputs.npm-auth-token }}
     - name: Install dependencies (npm)
-      if: steps.check_yarn_lock.outputs.files_exists == 'false'
+      if: steps.check_yarn_lock.outputs.files_exists == 'false' && steps.read_cache.outputs.cache-hit != 'true'
       shell: bash
       run: npm ci
       env:
         NPM_AUTH_TOKEN: ${{ inputs.npm-auth-token }}
+    - name: Cache node_modules
+      if: inputs.s3-bucket-name != ''
+      uses: everpcpc/actions-cache@v1
+      id: write_cache
```

Comment:
Should this only happen if there wasn't a hit?

Reply:
good idea